contracts/src/v0.8/automation/upkeeps/LinkAvailableBalanceMonitor.sol

```diff
@@ -267,19 +266,18 @@ contract LinkAvailableBalanceMonitor is AccessControl, AutomationCompatibleInter
       address targetAddress = targetAddresses[idx];
       contractToFund = s_targets[targetAddress];
       s_targets[targetAddress].lastTopUpTimestamp = uint56(block.timestamp);
-      if (
-        localBalance >= contractToFund.topUpAmount &&
-        _needsFunding(
-          targetAddress,
-          contractToFund.lastTopUpTimestamp + minWaitPeriod,
-          contractToFund.minBalance,
-          contractToFund.isActive
-        )
-      ) {
-        bool success = i_linkToken.transfer(targetAddress, contractToFund.topUpAmount);
+
+      (bool fundingNeeded, address target) = _needsFunding(
+        targetAddress,
+        contractToFund.lastTopUpTimestamp + minWaitPeriod,
+        contractToFund.minBalance,
+        contractToFund.isActive
+      );
+      if (localBalance >= contractToFund.topUpAmount && fundingNeeded) {
+        bool success = i_linkToken.transfer(target, contractToFund.topUpAmount);
         if (success) {
           localBalance -= contractToFund.topUpAmount;
-          emit TopUpSucceeded(targetAddress);
+          emit TopUpSucceeded(targetAddress, contractToFund.topUpAmount); // which address should be emitted??
```

Comment:
@RyanRHall what do u think of this address? the proxy address or the aggregator (the actual address which gets the LINK)?

Reply:
I like both those ideas :)